orangecontrib/text/tests/test_pubmed.py

```diff
@@ -0,0 +1,452 @@
+        list_input = [
+            'test',
+            'value',
+            'here',
+        ]
+
+        self.assertEqual(_join_if_list(not_list_input), not_list_input)
+        self.assertEqual(_join_if_list(list_input), 'test value here')
+
+    def test_mesh_headings_to_class(self):
+        input_headings = [
+            'heading1 & heading2/heading3,heading4/*heading5',
+            'heading1/heading2/*heading3',
+        ]
+        self.assertEqual(_mesh_headings_to_class(input_headings), 'heading1')
+
+    def test_date_to_iso(self):
+        # Correct inputs.
+        input_dates = [
+            '2015 Nov',
+            '2015',
+            '2015 Sep-Oct',
+            '2015 Fall',
+        ]
+
+        correct_results = [
+            '2015-11-01',
+            '2015-01-01',
+            '2015-09-01',
+            '2015-09-01',
+        ]
+
+        for date, result in zip(input_dates, correct_results):
+            self.assertEqual(_date_to_iso(date), result)
+
+        # Unexpected inputs.
+        unexpected_input = '2015 Unexpected'
+        self.assertWarns(
+            RuntimeWarning,
+            _date_to_iso,
+            unexpected_input,
+        )
+        self.assertEqual(_date_to_iso(unexpected_input), None)
+
+
+    def test_record_to_corpus(self):
+        mock_records = [
+            {
+                'FAU': ['Mock Author 1', 'Mock Author 2'],
+                'TI': 'Mock title',
+                'MH': ['heading1/heading2'],
+                'AB': 'Mock abstract',
+                'DP': '2015 Sep',
+            },
+        ]
+
+        correct_metas = np.array([
+            [
+                'Mock Author 1 Mock Author 2',
+                'Mock title',
+                'heading1/heading2',
+                'Mock abstract',
+                '2015-09-01'
+            ]
+        ])
+        correct_classes = np.array([
+            'heading1'
+        ])
+
+        # Perform asserting.
+        meta_values, class_values = _records_to_corpus_entries(
+                mock_records,
+                PUBMED_TEXT_FIELDS
+        )
+        corpus = _corpus_from_records(mock_records, PUBMED_TEXT_FIELDS)
+        self.assertCountEqual(meta_values[0], correct_metas[0])
+        self.assertCountEqual(class_values, correct_classes)
+        self.assertIsNotNone(corpus)
+
+    @patch('Bio.Entrez.esearch', mock_entrez.esearch)
+    @patch('Bio.Entrez.read', mock_entrez.read)
+    def test_pubmed_search_records(self):
+        test_terms = ['orchid']
+        authors = []
+        pub_date_start = '2011/07/07'
+        pub_date_end = '2014/07/07'
+
+        self.pubmed._search_for_records(
+                terms=test_terms,
+                authors=authors,
+                pub_date_start=pub_date_start,
+                pub_date_end=pub_date_end
+        )
+
+        # The only certain check is to make sure we got all the parameters.
+        self.assertIsNotNone(self.pubmed.record_id_list)
```

Comment:
Could you provide some more restricting checks. At least `self.assertIsInstance`?